lua/autorun/client/cl_remove_tabs.lua

```diff
@@ -30,33 +30,53 @@ local function hideTabs()
     end
 end
 
-hook.Add( "OnSpawnMenuOpen", "CFC_SpawnMenuWhitelist", function()
-    hook.Remove( "OnSpawnMenuOpen", "CFC_SpawnMenuWhitelist" )
+hook.Add( "SpawnMenuCreated", "CFC_SpawnMenuWhitelist", function()
     if LocalPlayer():IsAdmin() then return end
-
     hideTabs()
-    -- This extra pass ensures that all tabs are removed.
-    -- The Saves tab seems to be loaded in async so it wouldn't disappear until the second menu load
-    timer.Simple( 0, hideTabs )
 end )
 
 local emptyResults = function() return {} end
-local dupeError = "Workshop Dupes are disabled on this server"
 local errorSound = "buttons/button2.wav"
 
-hook.Add( "InitPostEntity", "CFC_SpawnMenuWhitelist", function()
-    if LocalPlayer():IsAdmin() then return end
+local function reject( message )
+    LocalPlayer():ChatPrint( message )
+    surface.PlaySound( errorSound )
+
+    return false
+end
 
+local function adjustSearchProfiders()
     -- Remove NPCs from the search provider
     local _, searchProviders = debug.getupvalue( search.AddProvider, 1 )
     searchProviders.npcs.func = emptyResults
+end
 
-    -- Disable workshop dupes only
+local function wrapDuplicator()
     ws_dupe._DownloadAndArm = ws_dupe._DownloadAndArm or ws_dupe.DownloadAndArm
-    ws_dupe.DownloadAndArm = function()
+
+    ws_dupe.DownloadAndArm = function( id )
+        -- Exploit fix?
         if not IsValid( LocalPlayer() ) then return end
-        LocalPlayer():ChatPrint( dupeError )
-        surface.PlaySound( errorSound )
+
+        -- Reject banned or non-owner dupes
+        steamworks.FileInfo( id, function( result )
```

Comment:
Looks the file info up first, then blocks banned (idk if that's possible, but I figure why not?) and dupes that aren't owned by LocalPlayer